src/less/star-rating-select/base/star-rating-select.less

```diff
@@ -0,0 +1,80 @@
+@import "../../mixins/icon/base/icon-mixins.less";
+@import "../../mixins/utility/utility-mixins.less";
+
+div.star-rating-select {
+    cursor: pointer; // this ensures text selection cursor is not used between stars
+    display: inline-block;
+}
+
+span.star-rating-select__radio {
+    display: inline-flex;
+    position: relative;
+    vertical-align: text-top;
+}
+
+span.span.star-rating-select__radio svg {
+    .icon-base-mixin();
+}
+
+span.star-rating-select__radio_icon {
+    display: inline-flex;
+    height: 14px;
+    outline-offset: 1px;
+}
+
+svg.star-rating-select__icon {
+    height: 12px;
+    width: 12px;
+}
+
+input.star-rating-select__radio_control[type="radio"] {
+    cursor: pointer;
+    font-size: 100%;
+    height: 12px;
+    margin: 0;
+    min-width: 12px;
+    opacity: 0;
+    padding: 0;
+    position: absolute;
+    width: 12px;
+    z-index: 1;
+}
+
+input.star-rating-select__radio_control[type="radio"]:focus + span.star-rating-select__radio_icon {
+    outline: 1px auto;
+    .outline-color-token(radio-outline, color-text-secondary);
+    outline-offset: 2px;
+}
+
+input.star-rating-select__radio_control[type="radio"]:focus:not(:focus-visible)
+    + span.star-rating-select__radio_icon {
+    outline: none;
+}
+
+div.star-rating-select svg.star-rating-select__icon {
+    --color-star-rating-full: var(
+        --color-star-rating-unfilled,
+        @color-star-rating-icon-unfilled-foreground
+    );
+}
+
+div.star-rating-select
+    input[type="radio"]:checked
+    + span.star-rating-select__radio_icon
+    svg.star-rating-select__icon {
+    --color-star-rating-full: var(
+        --color-star-rating-filled,
+        @color-star-rating-icon-filled-foreground
+    );
+}
+
+// trailing visual star selection star fills
+div.star-rating-select
+    input.star-rating-select_filled
+    + span.star-rating-select__radio_icon
+    svg.star-rating-select__icon {
+    --color-star-rating-full: var(
+        --color-star-rating-filled,
+        @color-star-rating-icon-filled-foreground
+    );
+}
```

Comment:
These can be combined
iv.star-rating-select
    input[type="radio"]:checked
    + span.star-rating-select__radio_icon
    svg.star-rating-select__icon,
div.star-rating-select
    input.star-rating-select_filled
    + span.star-rating-select__radio_icon
    svg.star-rating-select__icon {
// Styles here
}

Reply:
Ok. I was debating whether or not to do that. I kept them separate initially for readability but if that's not an issue, I'll combine them.

Reply:
Lets do it. I think it should be fine